merging-configmaps-gitops/README.md

```diff
@@ -0,0 +1,84 @@
+# Merging config in a gitops context
+
+Our app platform utilizes configmaps / secrets heavily to pass configuration into apps.
+
+*The following RFC will focus on configmaps but the same problem and solution are also applicable for secrets!*
+
+These configmaps are often either shared between multiple instances of the same app or have shared values between apps.
+In gitops the [intended solution](https://kubectl.docs.kubernetes.io/guides/introduction/kustomize/#2-create-variants-using-overlays) for both of these cases is to use `bases` and `overlays` which is achieved through `kustomize`.
+
+## Problem Statement
+
+The `data` section of a configmap is a mapping of keys to values.
+Therefore configuration (especially nested configuration) is usually grouped under a single key (e.g. `values`).
```

Comment:
We went with a single values key for simplicity when designing the config layers.

Adding the key name to the userConfig section similar to how Flux does this with the helmrelease CRD could help here. 

As suggested by @ubergesundheit in https://github.com/giantswarm/roadmap/issues/865


https://github.com/giantswarm/roadmap/issues/865